x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.test.ts

```diff
@@ -343,6 +336,26 @@ describe('endpoint list middleware', () => {
       expect(failedAction.error).toBe(apiError);
     });
 
+    it('should not call API again if it fails', async () => {
+      dispatchUserChangedUrl();
+
+      const apiError = new Error('oh oh');
+      const failedDispatched = waitForAction('endpointDetailsActivityLogChanged', {
+        validate(action) {
+          return isFailedResourceState(action.payload);
+        },
+      });
+
+      mockedApis.responseProvider.activityLogResponse.mockImplementation(() => {
+        throw apiError;
+      });
+
+      await failedDispatched;
+
+      expect(mockedApis.responseProvider.activityLogResponse).not.toHaveBeenCalledTimes(2);
```

Comment:
there's no need for this extra validation. The following line already covers this case.